### ISSUE TITLE
Move form related elements to @clayui/form package

### ISSUE THREAD

**Issue by bryceosterhaus** [liferay/clay#2137]
`RadioGroup`, `Checkbox`, `Autocomplete` are all components that should be rolled into the `@clayui/form` package since they are all form related html elements. Multi-select is another component that is going into `@clayui/form`.

Happy to have my opinion on this changed though.

**Comment by matuzalemsteles** [comment]
hey @bryceosterhaus I've been thinking more about this, maybe it makes sense to move `@clayui/date-picker`, `@clayui/color-picker` and `@clayui/time-picker` are essentially part of a Form and can be composed by others in the case of their input. What do you think?

**Comment by wincent** [comment]
As some level, all web apps *are* forms.  😂

A good rule of thumb for whether something should actually live inside the `form` package might be whether it corresponds, at least conceptually, to an HTML element that falls into [the "form content" category](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Form-associated_content); copy-pasting from that MDN page, the list is:

- `<button>`
- `<fieldset>`
- `<input>`
- `<keygen>`
- `<label>`
- `<meter>`
- `<object>`
- `<output>`
- `<progress>`
- `<select>`
- `<textarea>`

**Comment by bryceosterhaus** [comment]
> As some level, all web apps are forms.

I started to realize this with the PR I was working on. I agree in that we should probably only add html elements that fall into the form content category, however something like `<button>` makes sense to be a separate package to me, since often in a webapp we use it as an action and not necessarily always a form submission.

Overall, I would lean towards avoiding a hard and fast rule since we could probably end up just adding everything. For all of the picker components, I think I would opt to keep them separate since they are fairly complex. This nuance is the main reason I decided to start small and only [move checkbox and radio for now](https://github.com/liferay/clay/pull/2162)
